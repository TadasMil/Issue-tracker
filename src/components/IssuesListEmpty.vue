<script lang="ts" setup name="IssuesListEmpty"></script>

<template>
  <b-card
    text-variant="muted"
    title-tag="h2"
    align="center"
    class="shadow-sm border-0"
  >
    <div
      class="h-100 d-flex flex-column justify-content-center align-items-center"
    >
      <b-card-text>
        <h3 class="mb-1">No open issues found</h3>
      </b-card-text>
      <b-card-text>
        <p class="mb-1">Here you'll see open issues</p>
      </b-card-text>
    </div>
  </b-card>
</template>
