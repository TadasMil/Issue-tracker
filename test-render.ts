import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
