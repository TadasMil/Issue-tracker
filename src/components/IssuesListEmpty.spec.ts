import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import IssuesListEmpty from './IssuesListEmpty.vue'

describe('IssuesListEmpty', () => {
  it('should render the component', () => {
    const { html } = render(IssuesListEmpty)

    expect(html()).toMatchSnapshot()
  })
})
