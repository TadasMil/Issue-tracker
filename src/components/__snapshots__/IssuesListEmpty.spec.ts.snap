// Vitest Snapshot v1

exports[`IssuesListEmpty > should render the component 1`] = `
"<div class=\\"card text-center text-muted shadow-sm border-0\\">
  <!---->
  <!---->
  <div class=\\"card-body\\">
    <!---->
    <!---->
    <div class=\\"h-100 d-flex flex-column justify-content-center align-items-center\\">
      <p class=\\"card-text\\">
      <h3 class=\\"mb-1\\">No open issues found</h3>
      </p>
      <p class=\\"card-text\\">
      <p class=\\"mb-1\\">Here you'll see open issues</p>
      </p>
    </div>
  </div>
  <!---->
  <!---->
</div>"
`;
